src/hexo/themes/sonarwhal/source/js/scanner-common.js

```diff
@@ -37,25 +37,73 @@
         item.setAttribute('aria-expanded', 'false');
     };
 
-    var toggleExpand = function (evt) {
-        var element = evt.target;
+    var updateExpandAllButton = function (element, closeAll) {
+        var expanded = closeAll;
+
+        if (typeof expanded === 'undefined') {
+            var parent = element.closest('.rule-result');
```

Comment:
Maybe you can move this block of code to a new function `getStatus` or `someExpanded` or something similar